### PR TITLE
Feature - API tests with Auth

### PR DESCRIPTION
**Pull request by mgoodm**
- Mocks user authentication for controller tests in `PostControllerTest.cs`, `EditPostTests.cs`, `DeletePostTests.cs`, and `DeleteCommentTests.cs`
- Fix small bugs in controllers such as calling `CreatedAtAction` for Post methods
- Fix api route on the CommentsController `DeleteAsync` method to `api/comments/{commentId}`

### CODE REVIEW

**Review by nodeJayS — COMMENTED** [review]
Confirmed proper testing with okta, should prevent previous unit tests from failing since we enabled auth for controller action methods

**Comment by nodeJayS** [comment]
Note that this branch enables auth in all the controller methods that use it. This has to be the LAST branch to merge

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs [hunk 1]
```diff
@@ -41,17 +35,17 @@ ILogger<PostsController> logger
         [Authorize]
         [HttpPut("{id}")]
         public async Task<IActionResult> PutAsync(int id, Post post) {
-            /*try {
+            try {
                 var sessionEmail = User.FindFirst(ct => ct.Type.Contains("nameidentifier")).Value;
-                var postEmail = _postsRepository.First(p => p.Id == id).UserEmail;
+                var postEmail = _postsRepository.AsQueryable().First(p => p.Id == id).UserEmail;
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:40]: i'm not sure calling AsQueryable here does what you want it to do or not.

File: Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/DeleteCommentTests.cs [hunk 2]
```diff
@@ -13,7 +18,13 @@
 using Xunit;
 
 namespace Fakebook.Posts.UnitTests.Controllers {
-    public class DeleteCommentTests {
+    public class DeleteCommentTests : IClassFixture<WebApplicationFactory<Startup>> {
```
Comment (escalonn) [inline Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/DeleteCommentTests.cs:21]: all this stuff is integration tests, not unit tests, and so they don't belong in the unit test project and shouldn't contribute to code coverage statistics.
they're good though.

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs (hunk 1 above)
Comment (harelfishbein) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs:40]: To use linq method or their async versions in EntityFrameworkCore we need to call this on the repo, it doesn't pull any data just lets us call the extension methods like the AsEnumerable method.

File: Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/DeleteCommentTests.cs (hunk 2 above)
Comment (harelfishbein) [reply Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/DeleteCommentTests.cs:21]: So it is ok that we leave it for now to count toward the code coverage.

Comment (escalonn) [reply Fakebook.Posts/Fakebook.Posts.UnitTests/Controllers/DeleteCommentTests.cs:21]: yeah

File: Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/CommentsController.cs [hunk 3]
```diff
@@ -25,6 +24,41 @@ public class CommentsController : ControllerBase {
...
 
         /// <summary>
+        /// Deletes the post resource with the given id.
+        /// </summary>
+        /// <param name="id">Id of the post to be deleted</param>
+        /// <returns>An IActionResult containing either a:
+        /// 204NoContent on success,
+        /// 400BadRequest on delete failure,
+        /// 404NotFound if the Id did not match an existing post,
+        /// or 403Forbidden if the UserEmail on the original post does not match the email on the token of the request sender.</returns>
```
[the remark this answers is not in the document]
Comment (mgoodm) [reply Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/CommentsController.cs:34]: attributes have been added

**Comment by escalonn** [comment]
cf reply here https://github.com/2011-fakebook-project3/posts/pull/100#discussion_r556693180

**Review by RevatureGentry — APPROVED** [review]
LGTM

**Comment by nodeJayS** [comment]
Not enough time to test an async enumerator, requires changes on mocking the test

**Pull request merged**